www/content/faq.md

```diff
@@ -0,0 +1,447 @@
+to [Elm](https://elm-lang.org/)’s tangram logo.
+
+Roc is a direct descendant of Elm. The languages are similar, but not the same.
+[Origami](https://en.wikipedia.org/wiki/Origami) likewise has similarities to [tangrams](https://en.wikipedia.org/wiki/Tangram), although they are not the same.
+Both involve making a surprising variety of things
+from simple primitives. [_Folds_](<https://en.wikipedia.org/wiki/Fold_(higher-order_function)>)
+are also common in functional programming.
+
+The logo was made by tracing triangles onto a photo of a physical origami bird.
+It’s made of triangles because triangles are a foundational primitive in
+computer graphics.
+
+The name was chosen because it makes for a three-letter file extension, it means something
+fantastical, and it has incredible potential for puns. Here are some different ways to spell it:
+
+- **Roc** - traditional
+- **roc** - low-key
+- **ROC** - [YELLING](https://package.elm-lang.org/packages/elm/core/latest/String#toUpper)
+- **Röc** - [metal 🤘](https://en.wikipedia.org/wiki/Metal_umlaut)
+
+Fun fact: "roc" translates to 鹏 in Chinese, [which means](https://www.mdbg.net/chinese/dictionary?page=worddict&wdrst=0&wdqb=%E9%B9%8F) "a large fabulous bird."
+
+## Why can't functions be compared for equality using the `==` operator?
+
+Function equality has been proven to be undecidable in the general case because of the [halting problem](https://en.wikipedia.org/wiki/Halting_problem).
+So while we as humans might be able to look at `\x -> x + 1` and `\x -> 1 + x` and know that they're equivalent,
+in the general case it's not possible for a computer to do this reliably.
+
+There are some other potential ways to define function equality, but they all have problems.
+
+One way would be to have two functions be considered equal if their source code is equivalent. (Perhaps disregarding
+comments and spaces.) This sounds reasonable, but it means that now revising a function to do
+exactly the same thing as before (say, changing `\x -> x + 1` to `\x -> 1 + x`) can cause a bug in a
+distant part of the code base. Defining function equality this way means that revising a function's internals
+is no longer a safe, local operation - even if it gives all the same outputs for all the same inputs.
+
+Another option would be to define it using "reference equality." This is what JavaScript does, for example.
+However, Roc does not use reference equality anywhere else in the language, and it would mean that (for example)
+passing `\x -> x + 1` to a function compared to defining `fn = \x -> x + 1` elsewhere and then passing `fn` into
+the function might give different answers.
+
+Both of these would make revising code riskier across the entire language, which is very undesirable.
+
+Another option would be to define that function equality always returns `false`. So both of these would evaluate
+to `false`:
+
+- `(\x -> x + 1) == (\x -> 1 + x)`
+- `(\x -> x + 1) == (\x -> x + 1)`
```

Comment:
We could use roc syntax highlighting here now, so something like (note I've used `'` to stop it from rendering in this comment).
'''roc
(\x -> x + 1) == (\x -> 1 + x) # false
(\x -> x + 1) == (\x -> x + 1) # false
'''

Reply:
Ok, working on updating the rest of the formatting